modules/c/examples/code_snippets/main.c

```diff
@@ -1078,3 +1197,392 @@ Placeholder for Date accessors.
+    config.maxAttemptWaitTime = 600; //  <.>
+    // end::replication-set-maxattemptwaittime[]
+
+    // end::replication-retry-config[]
+
+    // tag::basic-authentication[]
+    // Configure Client Authentication to Basic Authentication
+    // Providing username and password credentials
+    if(docs_example_ShowBasicAuth) {
+        CBLAuthenticator* basicAuth =
+            CBLAuth_CreatePassword(FLSTR("username"),
+                                   FLSTR("passwd"));
+        config.authenticator = basicAuth; // <.>
+    }
+    // end::basic-authentication[]
+
+    // tag::session-authentication[]
+    if(docs_example_ShowSessionAuth) {
+        CBLAuthenticator* sessionAuth =
+            CBLAuth_CreateSession(FLSTR("904ac010862f37c8dd99015a33ab5a3565fd8447"),
+                                  FLSTR("optionalCookieName"));
+        config.authenticator = sessionAuth; // <.>
+    }
+
+    // end::session-authentication[]
+
+    // tag::replication-custom-header[]
+
+    // Optionally, add custom headers
+    FLMutableDict customHdrs = FLMutableDict_New();
+    FLMutableDict_SetString(customHdrs,
+                            FLSTR("customHeaderName"),
+                            FLSTR("customHeaderValue"));
+
+    config.headers = customHdrs;
+
+    // end::replication-custom-header[]
+
+    // FILTERS
+
+    // tag::replication-push-filter[]
+    // tag::replication-pull-filter[]
+    // Purpose - Illustrate use of push and-or pull filter functions
+
+    config.pushFilter = simpleReplicationFilter;
+
+    config.pullFilter = simpleReplicationFilter;
+
+    // end::replication-pull-filter[]
+    // end::replication-push-filter[]
+
+
+    //  Auto-purge over-ride
+    // tag::autopurge-override[]
+    config.disableAutoPurge = true; // <.>
+
+    // end::autopurge-override[]
+    // tag::[]
+
+    // Initialize replicator with created config
+    CBLReplicator* replicator =
+        CBLReplicator_Create(&config, &err); // <.>
+
+    // end::[]
+
+    CBLEndpoint_Free(target);
+
+    // Add optional change listener
+    CBLListenerToken* token_ReplChangeListener =
+        CBLReplicator_AddChangeListener(replicator,
+                                        simpleChangeListener,
+                                        NULL); // <.>
+
+
+}
+// END replication.html >> configure section
+
+
+
+
+// PAGE=Data Sync >> Initialize section
+// URL=https://docs.couchbase.com/couchbase-lite/current/c/replication.html#lbl-init-repl
+static CBLReplicator* docs_act_replication_Intialize(
+                        void* context,
+                        CBLReplicatorConfiguration argConfig,
+                        bool argResetRequired)
+{
+
+    CBLError err;
+
+    bool docs_example_resetRequired = argResetRequired;
+    // tag::p2p-act-rep-start-full[]
+
+    CBLReplicator* thisRepl =
+        CBLReplicator_Create(&argConfig, &err); // <.>
+
+    // end::p2p-act-rep-start-full[]
+    if(!docs_example_resetRequired) {
```

Comment:
What is all of this about?

Reply:
The conditionals again? Or the use of &addresses? Or the random use of tags?

Well the include picks up any content in the named tag block, or blocks; so to answer the following query the combined p2p-act-rep-start-full[] snippet does include the start.
What it doesn't include as (confusingly) structured here is the conditional statement syntax (see previous comment about pseudo-conditionals.
I guess I've not decided on approach yet (it's a beta her too).

BTW -- As of the 3.0 our snippets will include a couple click-thru to the github code; so that may change how we present them. Would be happy to get your ideas, Blake's too on how to improve that to better suit engineers.

Reply:
Perhaps I should explain that I've left every comment directly below the line I am commenting on unless stated otherwise :).  I'd be wary of clickthrough or presenting this as actual functioning code without testing it first.  It was designed to compile and look good on a page, and that was the extent of the requirement.